### ISSUE TITLE
Keyboard navigation, and onBlur not working on timepicker only

### ISSUE THREAD

**Issue by WGillian** [Hacker0x01/react-datepicker#2276]
**Describe the bug**
I am unable to select a time from the timepicker list using keyboard navigation. I can navigate away from the timepicker using TAB, but the timepicker stays open and the onBlur function is not called.

**To Reproduce:**
1) Create a timepicker using the `showTimeSelect` and `showTimeSelectOnly` props (or use the example select time only demo here https://reactdatepicker.com/#example-select-time-only)
2) Add `onBlur={() => console.log('onBlur event fired')}` and open console to see if onBlur event is fired
3) Focus input
4) Press tab

**Expected behavior:**
Focus appears on timepicker and I am able to use up and down arrows to select a time.
I can then click either enter or esc and focus returns to input and timepicker closes.
Clicking tab focusses next input and onBlur function is called.

**Actual behavior:**
Focus is lost from input and we can't see where it has gone.
Clicking tab again we still can't see where the focus is.
Clicking tab for a third time, focus appears on next input, time picker stays open, and onBlur function is never called.

**Desktop:**
- OS: mac OS High Sierra
- Browser: Chrome
- Version: 84

**Additional context**
I have based the above expected behaviour on the behaviour of the datepicker. 

Thanks

**Comment by WGillian** [comment]
For anyone looking for a temporary workaround, I did
`
const onTimepickerRef = timepickerRef => {
    if (timepickerRef && timepickerRef.input) {
      timepickerRef.input.onblur = () => timepickerRef.setOpen(false);
    }
  };
...
<DatePicker
              selected={date}
              onChange={onChangeTime}
              showTimeSelect
              showTimeSelectOnly
              ref={ref => onTimepickerRef(ref)}
              enableTabLoop={false}
 />
`
The key elements here being `enableTabLoop={false}` which stops focus going on the timepicker, and adding `timepickerRef.setOpen(false)` to the input's onblur event which closes the timepicker onblur.

This does not fix the issue, but allows keyboard users to manually enter the time and provides non-confusing user experience

**Comment by WGillian** [comment]
This is fixed in v3.4.0 with some cool keyboard navigation on the timepicker

**Comment by Vini-Dev** [comment]
My solution:

```jsx
    import { useRef } from 'react';
    import ReactDatePicker from 'react-datepicker';
    import 'react-datepicker/dist/react-datepicker.css';

    const datePickerRef = useRef(null);

    const handleOnKeyDown = (event) => {
      if (datePickerRef.current) {
        if (event?.keyCode === 9 || event?.which === 9) {
          datePickerRef.current?.setOpen(false);
        }
      }
    };
    
    return <ReactDatePicker ref={datePickerRef} onKeyDown={handleOnKeyDown} />;
```

**Comment by avi-dwave** [comment]
@WGillian I am using v3.6.0 but I don't get keyboard navigation when showTimeSelectOnly is true

**Comment by aalices** [comment]
Controlling if the dropdown is open or closed is possible like @joaoromeira suggested but the keyboard navigation itself is quite bad. I'd assume tak after the dropdown is opened I can navigate through options with up/down arrows but I can't. Also, hitting tab navigates to the last option in the list. Could this be improved?
